syne_tune/optimizer/schedulers/searchers/searcher.py

```diff
@@ -595,15 +658,43 @@ def _generate_remaining_candidates(self) -> List[dict]:
         return remaining_candidates
 
     def _generate_all_candidates_on_grid(self) -> List[dict]:
-        # Get hp values that are specified as Domain
+        """
+        Generates all configurations to be evaluated by placing a regular, equally spaced grid over the configuration space
+        Returns:
+            The set of all configurations on grid.
+        """
         hp_keys = []
         hp_values = []
-        for key, hp_range in reversed(list(self.config_space.items())):
-            if isinstance(hp_range, Domain):
-                hp_keys.append(key)
+        ## adding categorical, finiteRange, scalar parameters
+        for hp, hp_range in reversed(list(self.config_space.items())):
+            if isinstance(hp_range, Float) or isinstance(hp_range, Integer):
+                continue
+            if isinstance(hp_range, Categorical):
+                hp_keys.append(hp)
                 hp_values.append(hp_range.categories)
+            elif isinstance(hp_range, FiniteRange):
+                hp_keys.append(hp)
+                hp_values.append(hp_range.values)
+            elif not isinstance(hp_range, Domain):
+                hp_keys.append(hp)
+                hp_values.append([hp_range])
+
+        ## adding float, integer parameters
+        for hpr in self._hp_ranges._hp_ranges:
+            if hpr.name not in hp_keys:
+                _hpr_nsamples = self.num_samples[hpr.name]
+                _normalized_points = [
+                    (idx + 0.5) / _hpr_nsamples for idx in range(_hpr_nsamples)
+                ]
+                _hpr_points = [
+                    hpr.from_ndarray(np.array([point])) for point in _normalized_points
+                ]
+                _hpr_points = list(set(_hpr_points))
+                hp_keys.append(hpr.name)
+                hp_values.append(_hpr_points)
 
         hp_values_combinations = list(product(*hp_values))
+
         if self._shuffle_config:
             shuffle(hp_values_combinations)
```

Comment:
Sorry, I noted another miss in the current code.
This shuffle here uses a random generator. We'd like to control these. Could you please inherit from SearcherWithRandomSeed and make sure that this shuffle here uses the random_state? Check RandomSearcher for how this works.

Reply:
Matthias, Im not super clear on this one. I inherited from SearcherWithRandomSeed and used random_state in the shuffle as following:

import random
random.seed(self.random_state)
shuffle(self.hp_values_combinations)

is this what you meant?

Reply:
You can just do `self.random_state.shuffle(...)`